### PR TITLE
YAML guide

### LINKED ISSUE #132: Tutorial on YAML syntax
New users sometimes struggle with YAML syntax (maps, lists, block style vs flow style, multi-line string values, when to use quotes, etc).  The user guide should have a documentation page that summarizes the elements of YAML that are valid in CWL (and avoid covering features like tags and references that should not be used).

Inspired by conversation with @vzhz

### PR DESCRIPTION
**Pull request by tobyhodges**
Fixes #132 

Add a short tutorial for the subset of YAML features used in CWL.

### CODE REVIEW

**Comment by mr-c** [comment]
idea to include? `"1234"` is not the same as `1234`

File: _extras/yaml.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+it can severely affect the readability of a YAML file
+and should be used sparingly.
+
+### Reference
+
+The [Learn YAML in Y Minutes][yaml-y-mins] reference was very helpful for us
+while we wrote this guide.
+If you'd like to learn about some of the other features of YAML,
+which aren't valid for use with CWL,
+we highly recommend that you read it.
```
Comment (mr-c) [inline _extras/yaml.md]: I would remove this sentence. To the previous sentence I would add `, though it explains features of YAML that are not valid for CWL`

File: _extras/yaml.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+Values may be wrapped in quotation marks
+but be aware that this may change the way that they are interpreted
+i.e. `"1234"` will be treated as a character string
+, while `1234` will be treated as an integer.
+This distinction can be important,
+for example when describing parameters to a command:
+in CWL all parts of `baseCommand` must be strings so,
+if you want to specify a fixed numeric value to a command,
+make sure that you wrap that numeric value in quotes.
```
Comment (mr-c) [inline _extras/yaml.md]:
```suggestion
make sure that you wrap that numeric value in quotes: `baseCommand: [echo, "42"]`.
```

File: _episodes/02-1st-example.md [hunk 3]
```diff
@@ -14,7 +14,12 @@ keypoints:
 - "The tool description and input files are provided as arguments to a CWL runner."
 ---
 The simplest "hello world" program.  This accepts one input parameter, writes a message to the terminal or job log, and produces
-no permanent output. CWL documents are written in [JSON][json] or [YAML][yaml], or a mix of the two.
+no permanent output.
+CWL documents are written in [JSON][json] or [YAML][yaml-homepage], or a mix of the two.
+We will use YAML throughout this guide.
+If you are not familiar with this markup language,
```
Comment (mr-c) [inline _episodes/02-1st-example.md]:
```suggestion
If you are not familiar with YAML,
```

File: _extras/yaml.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+When describing a tool or workflow with CWL,
+it is usually necessary to construct more complex, nested representations.
+Called _maps_,
+these hierarchical structures are described in YAML by providing
+additional key-value pairs as the value of any key.
+These pairs (sometimes referred to as "children") are written
+on new lines under the key to which they belong (the "parent"),
+and should be indented with two spaces
+(tab characters are not allowed).
```
Comment (mr-c) [inline _extras/yaml.md]:
```suggestion
(⇥tab characters are not allowed).
```

**Comment by vzhz** [comment]
Glad to see this, thank you! 🙌

**Pull request merged**